SerialPrograms/Source/PokemonLA/Programs/PokemonLA_RegionNavigation.cpp

```diff
@@ -380,6 +380,94 @@ void goto_camp_from_overworld(
     context.wait_for(std::chrono::milliseconds((uint64_t)(GameSettings::instance().POST_WARP_DELAY * 1000)));
 }
 
+void goto_any_camp_from_overworld(ProgramEnvironment& env, ConsoleHandle& console, BotBaseContext& context,
+                                  const TravelLocation& location){
+
+    auto start = current_time();
+    std::chrono::seconds grace_period(0);
+    while (true){
+        {
+            EscapeFromAttack session(
+                env, console, context,
+                grace_period, std::chrono::seconds(10)
+            );
+            session.run_session();
+        }
+
+        if (current_time() - start > std::chrono::seconds(60)){
+            dump_image(console.logger(), env.program_info(), "EscapeFromAttack", console.video().snapshot());
+            throw OperationFailedException(console, "Unable to escape from being attacked.");
+        }
+
+        //  Open the map.
+        pbf_press_button(context, BUTTON_MINUS, 20, 30);
+        {
+            MapDetector detector;
+            int ret = wait_until(
+                console, context,
+                std::chrono::seconds(5),
+                {{detector}}
+            );
+            if (ret < 0){
+                dump_image(env.logger(), env.program_info(), "MapNotFound", console.video().snapshot());
+                throw OperationFailedException(console, "Map not detected after 5 seconds.");
+            }
+            console.log("Found map!");
+            context.wait_for(std::chrono::milliseconds(500));
+        }
+
+        //  Warp to sub-camp.
+        pbf_press_button(context, BUTTON_X, 20, 30);
+        {
+            ButtonDetector detector(
+                console, console,
+                ButtonType::ButtonA,
+                {0.55, 0.40, 0.20, 0.40},
+                std::chrono::milliseconds(200), true
+            );
+            int ret = wait_until(
+                console, context,
+                std::chrono::seconds(2),
+                {{detector}}
+            );
+            if (ret >= 0){
+                console.log("Flying back to camp...");
+                pbf_wait(context, 50);
+                if (location.warp_slot != 0){
```

Comment:
I don't think this does what you think it does. `warp_slot` is the slot when viewed from Jubilife. `warp_subslot` is the slot when viewed from inside the region. So for example, arenas and the Diamond/Pearl settlements are accessible via `warp_subslot`, but not via `warp_slot` from Jubilife.

Reply:
huum not sure, I got the idea of warp_slot being for the camps (accessible from Jubilife) and the subslot for the remaining locations. But since I'm using ```TravelLocations::instance()``` to travel, it won't work without that first block as it checks the first integer in the TravelLocation

eg. If I remove 
```
if (location.warp_slot != 0){
                    for (size_t c = 0; c < location.warp_slot; c++){
                        pbf_press_dpad(context, DPAD_DOWN, 20, 30);
                    }
                }
```

and try to go to Fieldlands_Heights (1,0), it will not work, it will only take in consideration the 0 (since the 1 was not used)

Reply:
Oh yeah you're right. I forgot that the sub-warp-slot is only when you've arrive.